### PR TITLE
sql: add switch for check constraint

### PR DESCRIPTION
**Pull request by fzzf678**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add a switch

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v7.2 (TiDB 7.2 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
https://github.com/pingcap/tidb/issues/41711
### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by fzzf678** [comment]
@CbcWestwolf  please take a tech review

File: system-variables.md
Comment (qiancai) [inline system-variables.md:1]: 因为此变量是默认关闭的，需要在 https://docs.pingcap.com/zh/tidb/dev/constraints#check-%E7%BA%A6%E6%9D%9F 文档里说明需要先开启此变量

**Review by TonsnakeLin — APPROVED** [review]
LGTM after fixing it by the comment

File: system-variables.md
Comment (fzzf678) [reply system-variables.md:1]: updated

File: constraints.md [hunk 1]
```diff
@@ -51,6 +51,10 @@ Query OK, 1 row affected (0.03 sec)
 
 ## `CHECK` 约束
 
+> **注意：**
+>
+> 为了和之前的版本保持兼容，`CHECK` 约束功能默认是关闭的，需要将 [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint) 开启后才能使用。
```
Comment (qiancai) [inline constraints.md:56]:
```suggestion
> `CHECK` 约束功能默认关闭，需要将 [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint) 开启后才能使用。
```

File: constraints.md [hunk 2]
```diff
@@ -128,6 +132,10 @@ ALTER TABLE t DROP CONSTRAINT t_chk_1;
 ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
 ```
 
+### 与 MySQL 的兼容性
```
Comment (qiancai) [inline constraints.md:135]:
```suggestion
### 与 MySQL 的兼容性

```

File: constraints.md [hunk 3]
```diff
@@ -128,6 +132,10 @@ ALTER TABLE t DROP CONSTRAINT t_chk_1;
...
+1. 不支持使用 `ALTER TABLE t ADD COLUMN a CHECK(a > 0)` 添加列的同时添加 `CHECK` 约束，使用该语句时 TiDB 不会报错，但是会忽略 `CHECK` 约束，列会被添加成功。
```
Comment (qiancai) [inline constraints.md:136]:
```suggestion
- 不支持在添加列的同时添加 `CHECK` 约束 (例如，`ALTER TABLE t ADD COLUMN a CHECK(a > 0)`) ，否则只有列会被添加成功，TiDB 会忽略 `CHECK` 约束但不会报错。
```

File: constraints.md [hunk 4]
```diff
@@ -128,6 +132,10 @@ ALTER TABLE t DROP CONSTRAINT t_chk_1;
...
+2. 不支持使用 `ALTER TABLE t CHANGE a b int CHECK(b > 0)` 添加 `CHECK` 约束，使用该语句时 TiDB 会报错。
```
Comment (qiancai) [inline constraints.md:137]:
```suggestion
- 不支持使用 `ALTER TABLE t CHANGE a b int CHECK(b > 0)` 添加 `CHECK` 约束，使用该语句时 TiDB 会报错。
```

File: system-variables.md [hunk 5]
```diff
@@ -1450,6 +1450,14 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 默认值：`OFF`
 - 这个变量用于控制是否开启 cascades planner。
 
+### `tidb_enable_check_constraint` <span class="version-mark">从 v7.2 版本开始引入</span>
```
Comment (qiancai) [inline system-variables.md:1453]:
```suggestion
### `tidb_enable_check_constraint` <span class="version-mark">从 v7.2.0 版本开始引入</span>
```

File: constraints.md [hunk 6]
```diff
@@ -51,6 +51,10 @@ Query OK, 1 row affected (0.03 sec)
 
 ## `CHECK` 约束
 
+> **注意：**
+>
+> 为了和之前的版本保持兼容，`CHECK` 约束功能默认是关闭的，需要将 [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint) 开启后才能使用。
+
```
Comment (qiancai) [inline constraints.md:57]: 是才能使用，还是才能生效呢

File: system-variables.md [hunk 7]
```diff
@@ -1450,6 +1450,14 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
...
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 该变量用于控制是否启用[`CHECK` 约束](/constraints.md#check-约束)。
```
Comment (qiancai) [inline system-variables.md:1459]:
```suggestion
- 这个变量用于控制是否启用[`CHECK` 约束](/constraints.md#check-约束)。
```

File: constraints.md (hunk 6 above)
Comment (fzzf678) [reply constraints.md:57]: 之前版本的行为是：可以使用(语法支持)但是不生效。当前版本的默认行为和之前版本一样，都是可以使用但是不会生效。

Comment (qiancai) [reply constraints.md:57]: 那改成“才能生效”？

File: constraints.md [hunk 8]
```diff
@@ -51,6 +51,10 @@ Query OK, 1 row affected (0.03 sec)
 
 ## `CHECK` 约束
 
+> **注意：**
+>
+> `CHECK` 约束功能默认关闭，需要将 [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-从-v720-版本开始引入) 开启后才能使用。
```
Comment (qiancai) [inline constraints.md:56]:
```suggestion
> `CHECK` 约束功能默认关闭，需要将 [`tidb_enable_check_constraint`](/system-variables.md#tidb_enable_check_constraint-从-v720-版本开始引入) 开启后才能生效。
```

File: system-variables.md [hunk 9]
```diff
@@ -1450,6 +1450,14 @@ mysql> SELECT job_info FROM mysql.analyze_jobs ORDER BY end_time DESC LIMIT 1;
 - 默认值：`OFF`
 - 这个变量用于控制是否开启 cascades planner。
 
+### `tidb_enable_check_constraint` <span class="version-mark">从 v7.2.0 版本开始引入</span>
+
+- 作用域：GLOBAL
+- 是否持久化到集群：是
+- 类型：布尔型
+- 默认值：`OFF`
+- 这个变量用于控制是否启用[`CHECK` 约束](/constraints.md#check-约束)。
```
Comment (Oreoxmt) [inline system-variables.md:1459]:
```suggestion
- 这个变量用于控制是否启用 [`CHECK` 约束](/constraints.md#check-约束)。
```

File: constraints.md [hunk 10]
```diff
@@ -128,6 +132,11 @@ ALTER TABLE t DROP CONSTRAINT t_chk_1;
 ALTER TABLE t ALTER CONSTRAINT c1 NOT ENFORCED;
 ```
 
+### 与 MySQL 的兼容性
+
+- 不支持在添加列的同时添加 `CHECK` 约束 (例如，`ALTER TABLE t ADD COLUMN a CHECK(a > 0)`) ，否则只有列会被添加成功，TiDB 会忽略 `CHECK` 约束但不会报错。
```
Comment (Oreoxmt) [inline constraints.md:137]:
```suggestion
- 不支持在添加列的同时添加 `CHECK` 约束（例如，`ALTER TABLE t ADD COLUMN a CHECK(a > 0)`)），否则只有列会被添加成功，TiDB 会忽略 `CHECK` 约束但不会报错。
```

**Review by Oreoxmt — APPROVED** [review]
/lgtm

**Review by qiancai — APPROVED** [review]

**Pull request merged**